# HG changeset patch
# User John W. Eaton <jwe@octave.org>
# Date 1505399797 14400
#      Thu Sep 14 10:36:37 2017 -0400
# Node ID 840882c82e22e067f06d29113ca6b8692243ee73
# Parent  1d5c85655a8ff433a8498299b626dfcd4006186a
install nested local functions only once (bug #51993)

* oct-parse.in.yy (base_parser::finish_function): Don't install nested
local functions as top-level local functions.

diff --git a/libinterp/parse-tree/oct-parse.in.yy b/libinterp/parse-tree/oct-parse.in.yy
--- a/libinterp/parse-tree/oct-parse.in.yy
+++ b/libinterp/parse-tree/oct-parse.in.yy
@@ -3468,19 +3468,20 @@ namespace octave
               {
                 fcn->mark_as_subfunction ();
                 m_subfunction_names.push_back (nm);
 
                 m_primary_fcn_scope->install_subfunction (nm, ov_fcn);
                }
           }
 
-        if (m_parsing_local_functions )
+        if (m_parsing_local_functions && m_curr_fcn_depth == 1)
           symtab.install_local_function (nm, octave_value (fcn), file);
-        else if (m_curr_fcn_depth == 1)
+
+        if (m_curr_fcn_depth == 1)
           fcn_scope->update_nest ();
 
         if (! m_lexer.reading_fcn_file && m_curr_fcn_depth == 1)
           {
             // We are either reading a script file or defining a function
             // at the command line, so this definition creates a
             // tree_function object that is placed in the parse tree.
             // Otherwise, it is just inserted in the symbol table,
